### ISSUE TITLE
Mouse button bindings are different with gpu-context=wayland(vk)

### ISSUE THREAD

**Issue by mielouk** [mpv-player/mpv#6796]
### mpv version and platform

1:0.29.1-6 on Arch with swayWM

### Reproduction steps
Set 
#MOUSE_BTN0 ignore                      # don't do anything
#MOUSE_BTN0_DBL cycle fullscreen        # toggle fullscreen on/off
#MOUSE_BTN2 cycle pause                 # toggle pause on/off
MOUSE_BTN3 add volume 2 #wheel up
MOUSE_BTN4 add volume -2 #wheel do
MOUSE_BTN5 seek -5 #wheel le 
MOUSE_BTN6 seek  5 #wheel ri
MOUSE_BTN8 seek  60 #for
MOUSE_BTN7 seek -60 #back

in input.conf

and

gpu-context=wayland
or
gpu-context=waylandvk

in mpv.conf

### Expected behavior

same as with X11 in input.conf

### Actual behavior
Thumb back - pause
Thumb for - pause
Wup - 3%
Wdo - 3%
Wle   - 10s
Wri - 10s

mpv --input-test --idle --force-window lists the available options. 
Nr. 1 is builtin default, but doesn't behave that way
Nr. 2 the actual input.conf line,  but doesn't behave that way
Nr. 3 'OSD keybindings, but doesn't behave that way

### Log file

https://8n1.org/15591/8905

### Sample files

any - https://www.youtube.com/watch?v=rTeLIFru-Eg

**Comment by Dudemanguy911** [comment]
I suppose it's not unthinkable that button names end up being different on wayland than on x11. Are you absolutely sure that you have the correct names for all of the buttons here? For me (I'm also on sway), wheel up and wheel down is `WHEEL_UP` and `WHEEL_DOWN`. Granted, my mouse is not fancy at all, but I don't see why the name would be different for a mouse with a bazillion different buttons. `mpv --input-test --idle --force-window --gpu-context=wayland` should tell you the correct button names in the wayland context.

**Comment by mielouk** [comment]
![Wire 2019-07-10 at 21_46](https://user-images.githubusercontent.com/3057062/60999866-72cf0780-a35c-11e9-9d8f-7765a5f29622.jpeg)

**Comment by mielouk** [comment]
This is one of the thumb buttons.

**Comment by Dudemanguy911** [comment]
The backward thumb button being detected as `MBTN_RIGHT` definitely strikes me as odd.

> The forward button shows the same, only _keybinding1 instead of 5.
Are you saying that it's also being detected as `MBTN_RIGHT` or is it just mapped to cycling pause as well?

**Comment by mielouk** [comment]
Both thumb buttons register as MBTN_RIGHT, as does the right button.

The amount the other values  change is also not right.

**Comment by Dudemanguy911** [comment]
It's plausible this is actually a wlroots and/or libinput issue. I can attempt to find a mouse somewhere later that actually has at least some more buttons (thumb buttons at least) and see what happens. It's hard to say for sure at this point.

**Comment by mielouk** [comment]
If it were the case, that it is a wlroots or libinput issue, shouldn't have other wayland driven apps the same or similar issues? All my other mouse bindings work just fine e.g. firefox, nemo ...

**Comment by Dudemanguy911** [comment]
Indeed, I finally got a mouse with more than three buttons and I can reproduce your problem with the thumb buttons (i.e. I got `MBTN_RIGHT` on releasing either buttons). It's also worth nothing that occasionally when pressing them I got something like `MBTN_RIGHT_DL` which I thought was interesting, but it didn't always appear. But you're right. Wayland firefox works fine, so this is indeed likely an mpv issue.
